python/kvikio/kvikio/remote_file.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+from __future__ import annotations
+
+import functools
+from typing import Optional
+
+from kvikio.cufile import IOFuture
+
+
+@functools.cache
+def is_remote_file_available() -> bool:
+    """Check if the remote module is available"""
+    try:
+        import kvikio._lib.remote_handle  # noqa: F401
+    except ImportError:
+        return False
+    else:
+        return True
+
+
+@functools.cache
+def _get_remote_module():
+    """Get the remote module or raise an error"""
+    if not is_remote_file_available():
+        raise RuntimeError(
+            "RemoteFile not available, please build KvikIO "
+            "with libcurl (-DKvikIO_REMOTE_SUPPORT=ON)"
+        )
+    import kvikio._lib.remote_handle
+
+    return kvikio._lib.remote_handle
+
+
+class RemoteFile:
+    """File handle of a remote file."""
+
+    def __init__(self, handle):
+        """Create a remote file from a Cython handle.
+
+        This constructor should not be called directly instead use a
+        factory method like `RemoteFile.open_http()`
+
+        Parameters
+        ----------
+        handle : kvikio._lib.remote_handle.RemoteFile
+            The Cython handle
+        """
+        assert isinstance(handle, _get_remote_module().RemoteFile)
+        self._handle = handle
+
+    @classmethod
+    def open_http(
+        cls,
+        url: str,
+        nbytes: Optional[int] = None,
+    ) -> RemoteFile:
+        """Open a http file.
+
+        Parameters
+        ----------
+        url
+            URL to the remote file.
+        nbytes
+            The size of the file. If None, KvikIO will ask the server
+            for the file size.
+        """
+        return RemoteFile(_get_remote_module().RemoteFile.open_http(url, nbytes))
+
+    def __enter__(self) -> RemoteFile:
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        pass
```

Comment:
Question. Should there be a `close` method that releases the handle, and then it is called in `__exit__`? This would kind of match usage of the builtin `open`.

Reply:
Agree, done